app/models/piece.rb

```diff
@@ -14,7 +14,73 @@ def black?
   def self.types
     %w(Pawn Rook Bishop Knight Queen King)
   end
-end
-
 
+  # is_obstructed? returns true if the current piece cannot move to its destination because a piece on its own team is in its way
+  def is_obstructed?(dest_x, dest_y)
+    # current location
+    current_x = x_position
+    current_y = y_position
+    # the change in state from current location to destination
+    delta_x = dest_x - current_x
+    delta_y = dest_y - current_y
 
+    # CHECK HORIZONTAL MOVES
+    # horizontal move where start is < destination and distance > 1
+    if delta_y == 0 && delta_x > 0 && delta_x.abs > 1
+      return Game.range_occupied?((current_x + 1), (dest_x - 1), current_y, current_y)
+    # horizontal move where start is > destination and distance > 1
+    elsif delta_y == 0 && delta_x < 0 && delta_x.abs > 1
+      return Game.range_occupied?((dest_x + 1), (current_x - 1), current_y, current_y)
+    # horizontal move to next square or delta == 0
+    elsif delta_y == 0 && delta_x.abs <= 1
+      return false
+    # CHECK VERTICAL MOVES
+    # vertical move where start is < destination and distance > 1
+    elsif delta_x == 0 && delta_y > 0 && delta_y.abs > 1
+      return Game.range_occupied?(current_x, current_x, (current_y + 1), (dest_y - 1))
+    # vertical move where start is > destination and distance > 1
+    elsif delta_x == 0 && delta_y < 0 && delta_y.abs > 1
+      return Game.range_occupied?(current_x, current_x, (dest_y + 1), (current_y - 1))
+    # vertical move to next square or delta == 0
+    elsif delta_x == 0 && delta_y.abs <= 1
+      return false
```

Comment:
Hey Charilie, I'm trying to understand these two lines, it seems as though there is no check to see if the adjacent vertical square is occupied.  I'm sure I've missed something, I'm just trying to understand why there's no need for Game.range_occupied? here.